mealie/services/migrations/cookn.py

```diff
@@ -0,0 +1,409 @@
+
+        # Get recipe image
+        _media_row = _media_recipe_row
+        _media_id = db.get_data(_media_row, "ID")
+        if _media_id == "":
+            # Get chaper image if no recipe image
+            _media_row = _media_chapter_row
+            _media_id = db.get_data(_media_row, "ID")
+        if _media_id == "":
+            # Get cookbook image if no chapter image
+            _media_row = _media_cookbook_row
+            _media_id = db.get_data(_media_row, "ID")
+
+        # If we found an image
+        if _media_id != "":
+            _media_type = db.get_data(_media_row, "MEDIA_CONTENT_TYPE")
+            # If the file has no extention add one (this is the normal case)
+            if Path(str(_media_id)).suffix == "":
+                if _media_type != "":
+                    # Determine file extension based on media type
+                    _extension = _media_type.split("/")[-1]
+                    _old_image_path = os.path.join(db.directory, str(_media_id))
+                    new_image_path = f"{_old_image_path}.{_extension}"
+                    # Rename the file if it exists and has no extension
+                    if os.path.exists(_old_image_path) and not os.path.exists(new_image_path):
+                        os.rename(_old_image_path, new_image_path)
+                    if Path(new_image_path).exists():
+                        return new_image_path
+            else:
+                return os.path.join(db.directory, str(_media_id))
+        return None
+
+    def _parse_ingredients(self, _recipe_id: str, db: DSVParser) -> list[RecipeIngredient]:
+        """Parses ingredients for recipe from Cook'n ingredients table."""
+        ingredients = []
+        ingredients_order = []
+        _ingredient_rows = db.query_by_id("temp_ingredient", "PARENT_ID", [_recipe_id])
+        for _ingredient_row in _ingredient_rows:
+            _unit_id = db.get_data(_ingredient_row, "AMOUNT_UNIT")
+            _unit_row = db.query_by_id("temp_unit", "ID", [_unit_id])[0]
+            _food_id = db.get_data(_ingredient_row, "INGREDIENT_FOOD_ID")
+            _food_row = db.query_by_id("temp_food", "ID", [_food_id])[0]
+            _brand_id = db.get_data(_ingredient_row, "BRAND_ID")
+            _brand_row = db.query_by_id("temp_brand", "ID", [_brand_id])[0]
+
+            amount_str = None
+            amount, _ = extract_quantity_from_string(db.get_data(_ingredient_row, "AMOUNT_QTY_STRING"))
+            if amount == 0:
+                amount = None
+                amount_str = db.get_data(_ingredient_row, "AMOUNT_QTY_STRING")
+            unit_name = db.get_data(_unit_row, "NAME")
+            food_name = db.get_data(_food_row, "NAME")
+
+            # Match unit and food from Mealie DB
+            unit = self.matcher.find_unit_match(unit_name)
+            if unit == "":
+                unit = None
+            food = self.matcher.find_food_match(food_name)
+            if food == "":
+                food = None
```

Comment:
These _should_ be returning only objects or `None`, which makes this redundant. Were you getting string results here? If so, might be a bug. If not, you can just remove the `if ... == ""` checks.